src/styled.js

```diff
@@ -71,10 +71,15 @@ const styled = ({tagName, elementStyle, mountSheet}: StyledArgs) => {
     updateSheet(props: StyledElementPropsType) {
       let rule
       let ruleIndex = 0
+
       // nested styles become to flatten rules, so we need to update each nested rule
       for (ruleIndex; ruleIndex < classMap[this.dynamicTagName].length; ruleIndex++) {
         rule = classMap[this.dynamicTagName][ruleIndex]
-        this.sheet.update(rule.name, props)
+
+        if (rule.name)
+          this.sheet.update(rule.name, props)
+        else
+          this.sheet.update(props)
```

Comment:
The problem here is that in this case we update the whole StyleSheet, and it will reduce performance significantly. That's why we update by the name, otherwise we can replace all that code with just `this.sheet.update` :)

But this solution should work:
```js
if (rule.name) this.sheet.update(rule.name, props)
if (rule.rules) rule.rules.update(props)
```

Conditional rules has their own `RulesContainer` in `rules`, so with `.update` we should iterate only for needed rules.